src-ghc/Pact/Server/PactService.hs

```diff
@@ -71,8 +71,12 @@ applyCmd logger conf dbv cv gasModel exMode _ (ProcSucc cmd) = do
       return cr
     Left e -> do
       logLog logger "ERROR" $ "tx failure for requestKey: " ++ show (cmdToRequestKey cmd) ++ ": " ++ show e
-      return $ jsonResult exMode (cmdToRequestKey cmd) $
-               CommandError "Command execution failed" (Just $ show e)
+      let
+        cValue :: CommandValue (Term Name)
+        cValue = CommandFailure $
+          CommandError "Command execution failed" (Just $ show e)
+      return $ jsonResult exMode (cmdToRequestKey cmd) cValue
```

Comment:
Semantic nitpick: can we move typed auxiliary functions to a `where` and throw a note on it so people know we're generalizing a phantom existential here? Thanks

Reply:
We've added the type annotation `CommandValue (Term Name)` to `cValue` because `CommandFailure a` needs to be typed as `CommandValue (Term Name)`, which is information that we cannot derive just from the values `CommandFailure` contains - ergo in the case of `CommandFailure`, the type parameter of `CommandValue` is a phantom type, and we're specializing to `Term name` on the fly.